2016/proposals/chris-medrela.md

```diff
@@ -0,0 +1,461 @@
+
+- Before introducing new features to all users, they'll be tested by a group of volunteering early users. That way, we don't have to test everything before showing it to first users, so we can get feedback faster and iterate quicker.
+
+- The early users can give feedback in an easy way, e.g. by sending an email rather than opening an issue on github.
+
+- UI mockups will be presented and accepted by the early users.
+
+- Amy dev instance will be available to all interested parties.
+
+It's also important to note what will be **not** implemented:
+
+- Advanced security control. There'll be only one flag in `Person` model that indicates if somebody is a Trainee with very limited permissions or somebody else (Admin, Trainer etc.) who can can edit everything.
+
+- Responsive interface.
+
+- Tracking sessions with Google Analytics or Piwik.
+
+## Milestone 1: Managing SWC Checkout Process
+
+That is everything after an Instructor Training -- that is Homework, Discussion Session, Checkout Session and welcoming to the community.
+
+The new workflow would be:
+
+- At the end of Instructor Training, Trainees register in Amy using the link shared by the Trainer. The link would be unique for each Training, so Trainees can be automatically associated with that Training.
+
+- The Amy interface for new trainees would be completely different from the current one. It could be a single page for the entire Checkout Process, plus one form to update their profile and one to change password.
+
+- Trainees make a change in the SWC Lessons and paste the link to their pull requests into Amy.
+
+- Amy analyzes the link to decide which core lesson did the trainee pick up.
+
+- After entering the pull request, the trainee uses Amy to register for one Discussion Session.
+
+    - In order not to overload Lesson Maintainers, they don't have to indicate in Amy if the pull request is good enough. It also means that trainees don't have to wait for the feedback from lesson maintainers, so they can register for discussion session immediately. This may raise up the percent of trainees who bother up to go through rest of checkout process.
+
+    - The Trainee can indicate that she's not available at any of the slots. She can tell when she's available in a text area. The Head of Training can list all trainees (in Trainees list view) who requested additional sessions along with whatever they entered in the text area.
+
+    - Amy tracks which Trainees attended which Sessions. That is, there would be many-to-many relation between Trainees and Discussion Sessions.
+
+- After the Discussion Session, the Discussion Leader uses the detailed view of Discussion Session shown below to indicate who has passed and who failed. After that, Amy mails the trainees with instructions on what to do next.
+![discussion-session-detailed-1](https://cloud.githubusercontent.com/assets/1645996/13723295/3d0e7582-e860-11e5-94ae-7e38a02c081a.png)
+
+- This view lets the Discussion Leader customize emails:
```

Comment:
Chris may not know about this, so let me partially answer that question: we have an email account for AMY to send out emails, we probably could use it in this instance too?